### ISSUE TITLE
ConvNeXt models missing gradients when compiling with `jit_compile=True`

### ISSUE THREAD

**Issue by ianstenbit** [keras-team/keras-cv#759]
Example error:

```
WARNING:tensorflow:Gradients do not exist for variables 
['convnext_xlarge_stage_0_block_0_depthwise_conv/kernel:0', 'convnext_xlarge_stage_0_block_1_depthwise_conv/kernel:0', 'convnext_xlarge_stage_0_block_2_depthwise_conv/kernel:0', 'convnext_xlarge_stage_1_block_0_depthwise_conv/kernel:0', 'convnext_xlarge_stage_1_block_1_depthwise_conv/kernel:0', 'convnext_xlarge_stage_1_block_2_depthwise_conv/kernel:0', 'convnext_xlarge_stage_2_block_0_depthwise_conv/kernel:0', 'convnext_xlarge_stage_2_block_1_depthwise_conv/kernel:0', 'convnext_xlarge_stage_2_block_2_depthwise_conv/kernel:0', 'convnext_xlarge_stage_2_block_3_depthwise_conv/kernel:0', 'convnext_xlarge_stage_2_block_4_depthwise_conv/kernel:0', 'convnext_xlarge_stage_2_block_5_depthwise_conv/kernel:0', 'convnext_xlarge_stage_2_block_6_depthwise_conv/kernel:0', 'convnext_xlarge_stage_2_block_7_depthwise_conv/kernel:0', 'convnext_xlarge_stage_2_block_8_depthwise_conv/kernel:0', 'convnext_xlarge_stage_2_block_9_depthwise_conv/kernel:0', 'convnext_xlarge_stage_2_block_10_depthwise_conv/kernel:0', 'convnext_xlarge_stage_2_block_11_depthwise_conv/kernel:0', 'convnext_xlarge_stage_2_block_12_depthwise_conv/kernel:0', 'convnext_xlarge_stage_2_block_13_depthwise_conv/kernel:0', 'convnext_xlarge_stage_2_block_14_depthwise_conv/kernel:0', 'convnext_xlarge_stage_2_block_15_depthwise_conv/kernel:0', 'convnext_xlarge_stage_2_block_16_depthwise_conv/kernel:0', 'convnext_xlarge_stage_2_block_17_depthwise_conv/kernel:0', 'convnext_xlarge_stage_2_block_18_depthwise_conv/kernel:0', 'convnext_xlarge_stage_2_block_19_depthwise_conv/kernel:0', 'convnext_xlarge_stage_2_block_20_depthwise_conv/kernel:0', 'convnext_xlarge_stage_2_block_21_depthwise_conv/kernel:0', 'convnext_xlarge_stage_2_block_22_depthwise_conv/kernel:0', 'convnext_xlarge_stage_2_block_23_depthwise_conv/kernel:0', 'convnext_xlarge_stage_2_block_24_depthwise_conv/kernel:0', 'convnext_xlarge_stage_2_block_25_depthwise_conv/kernel:0', 'convnext_xlarge_stage_2_block_26_depthwise_conv/kernel:0', 'convnext_xlarge_stage_3_block_0_depthwise_conv/kernel:0', 'convnext_xlarge_stage_3_block_1_depthwise_conv/kernel:0', 'convnext_xlarge_stage_3_block_2_depthwise_conv/kernel:0'] 
when minimizing the loss. If you're using `model.compile()`, did you forget to provide a `loss`argument?
```

**Comment by ianstenbit** [comment]
This is occurring when trying to train on TPU.

With `jit_compile=False`, training still fails, but with a different error:

```
tensorflow/core/tpu/kernels/tpu_compilation_cache_external.cc:113] Reshape's input dynamic dimension is decomposed into multiple output dynamic dimensions, but the constraint is ambiguous and XLA can't infer the output dimension %reshape.2192 = f32[64,56,56,1024]{3,2,1,0} reshape(f32[200704,1024]{1,0} %transpose.2191), metadata={op_type="Reshape" op_name="convnext_xlarge/convnext_xlarge_stage_0_block_0_pointwise_conv_1/Tensordot"}.
2022-09-02 17:21:54.799608: F tensorflow/core/tpu/kernels/tpu_program_group.cc:86] Check failed: xla_tpu_programs.size() > 0 (0 vs. 0)
```

**Comment by bhack** [comment]
Is it the same with `drop_path_rate=False`?

**Comment by ianstenbit** [comment]
@bhack yes, same error.

According to the error message above, it looks like the op with the shape error is `convnext_xlarge/convnext_xlarge_stage_0_block_0_pointwise_conv_1/Tensordot`, which is before the ConvNeXtBlocks where `drop_path_rate` is used.

**Comment by LukeWood** [comment]
@sayakpaul

**Comment by bhack** [comment]
Are you using

**Comment by sayakpaul** [comment]
Happens for all the variants or only XLarge?

**Comment by sayakpaul** [comment]
Need some reproducible code to debug it.

**Comment by ianstenbit** [comment]
@sayakpaul this is happening for all 3 that I've tested (Base, Large, XLarge)

**Comment by ianstenbit** [comment]
> Need some reproducible code to debug it.

I am running this using our ImageNet classification training script (with this [TPU training flag](https://github.com/keras-team/keras-cv/pull/757/files#diff-23eddb2c44a43ec109de63596e62d01ed6e8fd92eeb85c98a29fd7763ba76d07))

Invocation:
```
python3 -i ../keras-cv/examples/training/classification/imagenet/basic_training.py -imagenet_path=[redacted] --backup_path=[redacted] --weights_path=[redacted] --tensorboard_path=[redacted] --model_name=ConvNeXtXLarge --batch_size=512 --use_xla=False --use_tpu=True --model_kwargs="{'layer_scale_init_value': 1e-6, 'drop_path_rate': False}"
```

**Comment by sayakpaul** [comment]
A Colab Notebook is the only way for me to reproduce this. Could you provide one?

**Comment by ianstenbit** [comment]
> A Colab Notebook is the only way for me to reproduce this. Could you provide one?

I'm unable to get a Colab notebook set up running KerasCV at this time, as KerasCV requires TF 2.9, and Colab is currently running TF 2.8.

I will share a Colab notebook with this example once Colab is updated to a sufficient version of TF

**Comment by bhack** [comment]
@ianstenbit Ok, in the meantime, as you have the setup ready, can you test it with `drop_remainder=True`?

**Comment by ianstenbit** [comment]
> @ianstenbit Ok, in the meantime, as you have the setup ready, can you test it with `drop_remainder=True`?

Sorry I forgot to reply to this earlier -- I gave that a shot and it did not fix the problem.

**Comment by bhack** [comment]
> Sorry I forgot to reply to this earlier -- I gave that a shot and it did not fix the problem.

Thanks.

**Comment by sayakpaul** [comment]
> > A Colab Notebook is the only way for me to reproduce this. Could you provide one?
> 
> I'm unable to get a Colab notebook set up running KerasCV at this time, as KerasCV requires TF 2.9, and Colab is currently running TF 2.8.
> 
> I will share a Colab notebook with this example once Colab is updated to a sufficient version of TF

Appreciate it, thanks.

**Comment by sachinprasadhs** [comment]
Thanks for reporting the issue! We have consolidated the development of KerasCV into the new **KerasHub** package, which supports image, text, and multi-modal models. Please read [the announcement](https://github.com/keras-team/keras-hub/issues/1831). `KerasHub` will support all the core functionality of `KerasCV`.

`KerasHub` can be installed with `!pip install -U keras-hub`. Documentation and guides are available at [keras.io/keras_hub](https://keras.io/keras_hub/).

With our focus shifted to KerasHub, we are not planning any further development or releases in KerasCV. If you encounter a KerasCV feature that is missing from KerasHub, or would like to propose an addition to the library, please file [an issue with KerasHub](https://github.com/keras-team/keras-hub/issues).
